File: ChatWebApp/Entities/Conversation.cs
namespace ChatAppAPI.Entities
{
    public class Conversation: IAuditable
    {
        public Guid Id { get; set; }
        public string? Name { get; set; }
        public string? Avatar { get; set; }
        DateTime IAuditable.CreatedDate { get; set; }
        DateTime IAuditable.ModifiedDate { get; set; }
        User IAuditable.CreatedBy { get; set; }
        User IAuditable.ModifiedBy { get; set; }
    }
}

File: ChatWebApp/ClientApp/src/app/_models/conversation.ts
import { ConversationParticipant } from "./conversation-participant.model";

export class Conversation {
  id?: string;
  name?: string;
  avatar?: string;
  createdDate!: Date;
  modifiedDate?: Date;
  createdBy?: string;
  modifiedBy?: string;
}
export class ConversationDto {
  id?: string;
  name?: string;
  avatar?: string;
  lastMessage?: string;
  lastSender?: string;
  lastSent!: Date;
}
export class ConversationInfoDto{
  id?: string;
  name?: string;
  avatar?: string;
  participants?: ConversationParticipant[];
}
